Neos.Media.Browser/Classes/Controller/ImageController.php

```diff
@@ -28,14 +30,31 @@ class ImageController extends AssetController
     protected $assetRepository;
 
     /**
+     * @Flow\Inject
+     * @var \Neos\Media\Domain\Repository\ImportedAssetRepository
+     */
+    protected $importedAssetRepository;
+
+    /**
+     * @param string $assetSourceIdentifier
+     * @param string $assetProxyIdentifier
      * @param Asset $asset
-     * @return void
+     * @return void|string
+     * @throws \Neos\Flow\Mvc\Exception\StopActionException
+     * @throws \Neos\Flow\Mvc\Exception\UnsupportedRequestTypeException
      */
-    public function editAction(Asset $asset)
+    public function editAction(string $assetSourceIdentifier = null, string $assetProxyIdentifier = null, Asset $asset = null)
     {
-        if ($asset instanceof ImageVariant) {
-            $asset = $asset->getOriginalAsset();
+        if ($assetSourceIdentifier !== null && $assetProxyIdentifier !== null) {
+            parent::editAction($assetSourceIdentifier, $assetProxyIdentifier);
+            return;
```

Comment:
Shouldn't this actually `return parent::editAction($assetSourceIdentifier, $assetProxyIdentifier);` ?

Reply:
The parent `editAction` is declared as `@return void`, so if it was `return parent::editAction()…`, Phpstorm complains …

But right, it looks a bit suspicious.